mailpile/mail_source/__init__.py

```diff
@@ -163,6 +162,8 @@ def _save_config(self):
             self.session, 'Save config', self.session.config.save)
 
     def _log_status(self, message):
+        # If the user renames our parent_tag, we assume the name change too.
+        self.update_name_to_match_tag()
```

Comment:
> Well, the logging is what triggers UI updates (events, notifications) so ensuring we have the right idea about what our name is before doing so is reasonable. 

> But generally speaking, this is pure side-effect code and as such is ugly almost anywhere we put it. It's a hack, but without triggers on the settings themselves, it's hard to do this cleanly.

We do agree on the fact that side effects are bad. But given we have a side-effect method, let's not hide its call it in a method (`log_status`) that doesn't give a clue about the fact it will trigger side effect on the object itself.